### DISCUSSION TITLE
Is there a way to enforce createNamespace to be false?

### DISCUSSION THREAD

**Discussion by ehuizar1028** [helmfile/helmfile#1750] [Questions]
So i have a different process that creates namespaces that is guardrailed and standardized. I am using helmfile for the first time, and I am trying to prevent users from enabling `createNamespace: true`. All helm deploys go through Jenkins. 

So I put in a groovy function that would enforce this and inject `createNamespace: false` but it does not work if I set any templating since groovy cannot render it as yaml (using readYAML utility function)

Is there any easy way to set `createNamespace` to false for all releases?

**Comment by yxxhero** [comment]
@ehuizar1028 I didn't get you.

**Reply by ehuizar1028** [reply]
I would like to set `createNamespace: false` either through environment variable, or CLI argument. Is there a way to do that? 

I don't want to give me users the ability to create namespaces during helm deployments

**Comment by yxxhero** [comment]
@ehuizar1028 but  your usess can create namespace by create namespace yaml. how do you prevent that?

**Reply by ehuizar1028** [reply]
My use case isn't creating a namespace. 

My use case deploys an application to existing namespaces for developers. But I want to restrict users the ability of creating their own namespaces because I have a different process in Terraform that creates namespaces.

**Comment by yxxhero** [comment]
So I put in a groovy function that would enforce this and inject createNamespace: false but it does not work if I set any templating since groovy cannot render it as yaml (using readYAML utility function)


could  you give me some demo or logs?

**Reply by ehuizar1028** [reply]
To reproduce the issue. 

my jenkins groovy has something like this: 
```
helmFile = readyaml(file: "helmfile.yaml")
if(!helmFile.containsKey("helmDefaults")) {
  helmFile.helmDefaults = [:]
}
helmFile.helmDefaults["createNamespace"] = false
writeYaml data: helmFile, file:"helmFile.yaml"
```
Which does work to prevent users from creating their own namespace in our pipelines we manage. even if they attempt to define `createNamespace: true` themselves in their `helmfile.yaml`, we overwrite it to false. 

BUT if I have this in my helmfile.yaml, the `readYaml` function above errors because it cannot resolve the templating blocks
```
releases:
  {{- range $dir := readDirEntries "./" }}
  {{- if $dir.IsDir }}
  - name: appName
    chart: ChartName
  {{ -end }}
  {{ -end }}
```

if you're interested in the exact error
```
expected the node content, but found "-' in 'reader', line 34, column 5:
{{- range $dir := readDirEntries" ...
```

So my question is if there's a way to enforce `createNamespace: false` from either a CLI argument, environment variable, etc? I've also tried layering helmFile values and that also works but users can still create namespaces IF they manually specify in their `helmfile` `createNamespace: true"..
